hcl/hcl_import_test.go

```diff
@@ -0,0 +1,457 @@
+	"github.com/mineiros-io/terramate/hcl"
+)
+
+func TestHCLImport(t *testing.T) {
+	for _, tc := range []testcase{
+		{
+			name: "import with label - fails",
+			input: []cfgfile{
+				{
+					filename: "cfg.tm",
+					body: `import "something" {
+						source = "bleh"
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("cfg.tm", start(1, 8, 7), end(1, 19, 18))),
+				},
+			},
+		},
+		/*{
+			name: "import missing source - fails",
+			input: []cfgfile{
+				{
+					filename: "cfg.tm",
+					body: `import {
+
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrTerramateSchema,
+						mkrange("cfg.tm", start(1, 8, 7), end(1, 8, 18))),
+				},
+			},
+		},*/
+		{
+			name:     "import with non-existent file - fails",
+			parsedir: "stack",
+			input: []cfgfile{
+				{
+					filename: "stack/cfg.tm",
+					body: `import {
+						source = "/other/non-existent-file"
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrImport,
+						mkrange("stack/cfg.tm", start(2, 16, 24), end(2, 42, 50))),
+				},
+			},
+		},
+		{
+			name: "import same file - fails",
+			input: []cfgfile{
+				{
+					filename: "cfg.tm",
+					body: `import {
+						source = "cfg.tm"
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrImport,
+						mkrange("cfg.tm", start(2, 16, 24), end(2, 24, 32))),
+				},
+			},
+		},
+		{
+			name: "import same directory - fails",
+			input: []cfgfile{
+				{
+					filename: "cfg.tm",
+					body: `import {
+						source = "other.tm"
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrImport,
+						mkrange("cfg.tm", start(2, 16, 24), end(2, 26, 34))),
+				},
+			},
+		},
+		{
+			name:     "import cycle - fails",
+			parsedir: "stack",
+			input: []cfgfile{
+				{
+					filename: "stack/cfg.tm",
+					body: `import {
+						source = "/other/cfg.tm"
+				}`,
+				},
+				{
+					filename: "other/cfg.tm",
+					body: `import {
+						source = "/stack/cfg.tm"
+				}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrImport,
+						mkrange("other/cfg.tm", start(2, 16, 24), end(2, 31, 39))),
+				},
+			},
+		},
+		{
+			name:     "import same tree - fails",
+			parsedir: "stack",
+			input: []cfgfile{
+				{
+					filename: "stack/cfg.tm",
+					body: `import {
+						source = "/cfg.tm"
+				}`,
+				},
+				{
+					filename: "cfg.tm",
+					body: `terramate {
+					}`,
+				},
+			},
+			want: want{
+				errs: []error{
+					errors.E(hcl.ErrImport,
+						mkrange("stack/cfg.tm", start(2, 16, 24), end(2, 25, 33))),
+				},
+			},
+		},
+		{
+			name:     "import same file multiple times - fails",
+			parsedir: "stack",
+			input: []cfgfile{
+				{
+					filename: "stack/cfg.tm",
+					body: `import {
+						source = "/other/cfg.tm"
+				}
+				
+				import {
+					source = "/other/cfg.tm"
+			}`,
+				},
```

Comment:
```suggestion
			input: []cfgfile{
				{
					filename: "stack/cfg.tm",
					body: `
						import {
							source = "/other/cfg.tm"
						}
				
						import {
							source = "/other/cfg.tm"
						}
					`,
				},
```

My mind got a little fucked up because of the tabs here :laughing:

Reply:
oh no, this breaks the annoying file ranges I already suffered to come up with haha

Reply:
ah maybe at least fix the identation, or that would also break it (not adding newlines) ? Im making suggestions because overall the identation on these blocks are not very consistent and some versions of them are a little confusing/hard to understand (like this one). But we can improve this later too. Not a major issue.

Reply:
fixed